src/main/scala/temple/RegenerationFilter.scala

```diff
@@ -28,21 +27,19 @@ object RegenerationFilter {
 
   // Get all the files under a directory, up to a max depth of 4 (the deepest Temple generates)
   // TODO: Should this be moved to FileUtils?
-  private def filesUnderDirectory(directory: String): Seq[File] =
-    if (new ioFile(directory).exists()) {
+  private def filesUnderDirectory(directory: String): Seq[File] = {
+    val directoryPath = Paths.get(directory)
+    if (Files.exists(directoryPath)) {
       Files
-        .walk(FileSystems.getDefault.getPath(directory), 4)
+        .walk(directoryPath, 4)
         .iterator()
         .asScala
         .filterNot(path => path.toFile.isDirectory)
-        .map(path => FileSystems.getDefault.getPath(directory).resolve(path))
-        .map(path =>
-          // hack - if the file is in $directory then getParent wont contain "directory/", but it will when the
-          // file is in some subdirectory of $directory, allow for both cases
-          File(path.getParent.toString.replace(directory + "/", "").replace(directory, ""), path.getFileName.toString),
-        )
+        .map(path => directoryPath.relativize(path))
+        .map(path => File(Option(path.getParent).fold("")(_.toString), path.getFileName.toString))
         .toSeq
     } else Seq()
+  }
```

Comment:
How’s this different from `temple.TestUtils#buildFileMap`? Worth moving `buildFileMap` to FileUtils, and just using that?